resources/sass/button.scss

```
@@ -1,19 +1,140 @@
-.btn {
+@import "variables";
+@import "mixins";
+
+/* remove standard styles from links */
+.btn,
+.btn:active,
+.btn:hover {
+    text-decoration: none;
+}
 
-    @extend .btn;
-    box-shadow: $btn-box-shadow;
+/* remove standard styles from inputs and buttons */
+.btn {
+    outline: none;
+    border: none;
 
+    &:focus {
+        outline: none;
+    }
 }
 
-.btn:focus {
+/* variables and mixins used in the button styling */
+@mixin button-size($size: normal) {
+    /*
+     * Adds a padding for the button,
+     * according to given size.
+     * */
+
+    @if $size == small {
+        font-size: 0.8rem;
+        padding: 0.3rem;
 
-    @extend %form-content-focus;
+    } @else if $size == normal {
+        font-size: 1rem;
+        padding: 0.6rem;
 
+    } @else if $size == large {
+        font-size: 1.4rem;
+        padding: 0.9rem;
+    }
 }
 
-.btn:hover {
+$button-border-weight: 3px;
+
+/* standard button styles */
+.btn {
+    @include button-size(normal);
+    // display: flex;
+    // align-items: center;
+    // justify-content: space-between;
+
+    // text-transform: uppercase;
+
+    border-radius: $button-border-weight;
+
+    transition: all 0.2s ease;
+
+    text-align: center!important;
+    cursor: pointer;
+
+    /* button icon */
+    i {
+        vertical-align: middle;
+        padding: 0;
+        margin: 0;
+    }
+
+    /* button's shadows */
+    &--shadow {
+        @include box-shadow(sm);
+    }
+
+    &--shadow-md {
+        @include box-shadow(md);
+    }
+
+    &--shadow-lg {
+        @include box-shadow(lg);
+    }
+
+    /* full width button */
+    &--full {
+        display: inline-block;
+        width: 100%;
+    }
+
+    /* button sizes */
+    &--small {
+        @include button-size(small);
+    }
+
+    &--large {
+        @include button-size(large);
+    }
+
+    /* button theme variations */
+    @each $color, $value in $theme-colors {
+        &--#{$color} {
+            background-color: $value;
+            color: color-yiq($value);
+
+            &:hover {
+                background-color: theme-color-level($color, 2);
+            }
+        }
+
+        /* outline variant */
+        &--#{$color}-outline {
```

Sobre as classes {color}-outline:
Eu acredito que seria melhor reservar uma classe como apenas `.btn--outline`, e essa classe já deixará o botão com esse estilo independe da cor dele, a ideia é reservar de certa forma cada modificador para uma função.
Nesse sentido você pode fazer algo como:
```suggestion
        &--#{$color}.&--outline {
```

De fato vai ficar mais verboso, mas fica muito mais intuitivo no momento de atribuir as classes.